hlx_statics/blocks/herosimple/herosimple.js

```diff
@@ -1,49 +1,91 @@
+
+  block.setAttribute('daa-lh', 'hero');
+  block.querySelectorAll('h1, h2, h3, h4, h5, h6').forEach((h) => {
+    const fontFamily = block?.parentElement?.parentElement?.getAttribute('data-font-family');
+    const headerFontSize = block?.parentElement?.parentElement?.getAttribute('data-HeaderFontSize');
+    if (fontFamily) {
+      h.style.fontFamily = fontFamily;
+      h.classList.add('spectrum-Heading', 'spectrum-Heading--sizeXXL');
+    } else if (headerFontSize) {
+      h.classList.add('spectrum-Heading', 'spectrum-Heading--sizeXXL');
+      h.style.fontSize = headerFontSize;
+    } else {
+      h.classList.add('spectrum-Heading', 'spectrum-Heading--sizeXXL', 'spectrum-Heading');
+    }
+
+  });
+
+  // arrange divs
+  let firstDiv = block.firstElementChild;
+  let secondDiv = block.lastElementChild;
+
+  let wrapperDiv = createTag('div');
+  wrapperDiv.appendChild(firstDiv);
+  wrapperDiv.appendChild(secondDiv);
+
+  block.appendChild(wrapperDiv);
+
+  // second child inner div make it a p
+  let descriptionP = secondDiv.innerText;
+  let descriptionPElement = createTag('p', { class: 'spectrum-Body spectrum-Body--sizeL' });
+  descriptionPElement.innerText = descriptionP;
+
+  secondDiv.replaceWith(descriptionPElement);
+  // Paragraph decoration
+  block.querySelectorAll('p').forEach((p) => {
+    if (p.innerText) {
+      p.classList.add('spectrum-Body', 'spectrum-Body--sizeL');
+    }
+  });
+  const sourceElement = block.querySelector('source[type="image/webp"]');
+  const srcsetValue = sourceElement ? sourceElement.getAttribute('srcset') : null;
+  const url = srcsetValue.split(' ')[0];
+  const imgElement = block.querySelector('img');
+  const heroFirstDiv = block.querySelector('.herosimple > div');
+  const heroSecondDiv = block.querySelector('.herosimple > div:nth-of-type(2)');
+  if (srcsetValue) {
+    imgElement.style.display = 'none';
+    heroFirstDiv.style.marginBottom = '0px';
+    heroSecondDiv.style.marginTop = '0px';
+    Object.assign(block.style, {
+      backgroundImage: `url(${url})`,
+      backgroundSize: 'cover',
+      backgroundPosition: 'center',
+      backgroundRepeat: 'no-repeat'
     });
-  
-    // arrange divs
-    let firstDiv = block.firstElementChild;
-    let secondDiv = block.lastElementChild;
-  
-    let wrapperDiv = createTag('div');
-    wrapperDiv.appendChild(firstDiv);
-    wrapperDiv.appendChild(secondDiv);
-  
-    block.appendChild(wrapperDiv);
-  
-    // second child inner div make it a p
-    let descriptionP = secondDiv.innerText;
-    let descriptionPElement = createTag('p', {class:'spectrum-Body spectrum-Body--sizeL'});
-    descriptionPElement.innerText = descriptionP;
-  
-    secondDiv.replaceWith(descriptionPElement);
-    // Paragraph decoration
-    block.querySelectorAll('p').forEach((p) => {
-      if (p.innerText) {
-        p.classList.add('spectrum-Body', 'spectrum-Body--sizeL');
+  }
+
+  const sections = document.querySelectorAll('.section');
```

Comment:
This selects all of the sections in the entire document, rather than just the `herosimple` block. Ideally, code in `herosimple.js` only operates on the `herosimple` block.

An alternative, and hopefully more efficient way of selecting`herosimple-container` is to start from the `herosimple` block and move upwards like this: https://github.com/AdobeDocs/adp-devsite/blob/one-repo/hlx_statics/scripts/lib-adobeio.js#L149-L150

Reply:
Okay, @melissag-ensemble , I will take a look.